template/pyproject.toml.jinja

```diff
@@ -26,13 +26,32 @@ classifiers = [
 ]
 requires-python = ">={{min_python}}"
 
+dynamic = ["version"]
+
 # IMPORTANT:
 # Run 'tox -e deps' after making changes here. This will update requirement files.
 # Make sure to list one dependency per line.
 dependencies = [
 ]
 
-dynamic = ["version"]
+# Map from pip package names to conda package names
+# when the two are different.
+# Used for both package dependencies and test dependencies.
+# Must include optional dependency spec and all extra requirements
+# must be listed under `tool.conda_meta.extra_dependencies.run`
+# or `tool.conda_meta.extra_dependencies.test`
+# E.g.
+#  graphviz = "python-graphviz"
+#  "scipp[all]" = "scipp"
+[tool.conda_meta.dependency_map]
+
+# List any dependencies needed by the conda package
+# but not by the wheel or regular project tests.
+[tool.conda_meta.extra_dependencies]
+run = [
+]
+test = [
+]
```

Comment:
I think I would suggest removing all of this.

- The dependency map we can hard-code in `make_conda_meta.py`, since it is rare and we probably know all cases.
- If there are extra run and test dependencies, why should projects not add them in their `meta.yaml` directly? That would be least confusing for developers. The above requires a lot of non-standard stuff that a developer will run into. Can the `make_conda_meta.py` be modified so it leaves changes outside the auto-generated section intact?

Reply:
> Can the make_conda_meta.py be modified so it leaves changes outside the auto-generated section intact?

Not with pyyaml because it does not preserve formatting and comments. We could of course use jinja or some other templating engine. But we already have two level of template in this file (copier and conda).

The dependencies can be moved into meta_template.yaml and the script just extends the sections.section

> The dependency map we can hard-code in make_conda_meta.py, since it is rare and we probably know all cases.

Not if you want to keep Scipp-specifics to a minimum (#141). Then outside users would need a way to extend the map.

Reply:
`python-graphviz` is not really Scipp specific? 

Regarding the pyyaml, I still don't really see the case for having this in `pyproject.toml`. Can it move into the `meta` template? I am not sure if completely preventing any user-modification of either the `meta.yaml` or the template is feasible? If we want to be generic then we have to allow for some sort of modification outside of the possibilities that we have considered.

And finally, how much of this is still relevant if we want to move to `conda-forge`? Would we still have the `meta.yaml` here and then copy it to the conda-forge recipes?

Reply:
> python-graphviz is not really Scipp specific?

Yes. But the fact that we depend on it is. Other projects may depend on other packages that have different names in pip and conda.

> Regarding the pyyaml, I still don't really see the case for having this in pyproject.toml. Can it move into the meta template?

Regarding listing extra deps, I agree as I said. But I would not like encoding the map in the template. This would mean that part of the template is copied and extended to make `meta.yaml` and part of it is processed differently and omitted from `meta.yaml`.

> I am not sure if completely preventing any user-modification of either the meta.yaml or the template is feasible?

That's perfectly fine. The template can be modified freely.

> And finally, how much of this is still relevant if we want to move to conda-forge? Would we still have the meta.yaml here and then copy it to the conda-forge recipes?

It would still be possible to use a script. But it would have to write its output to a separate repo.
We would have to manage dependencies in the same way, conda-forge doesn't do that for us, AFAIK.